packages/core/src/codewhispererChat/tools/grepSearch.ts

```diff
@@ -0,0 +1,214 @@
+        updates.write(`Grepping for "${this.query}" in directory: ${searchDirectory}`)
+        updates.end()
+    }
+
+    public async invoke(updates?: Writable): Promise<InvokeOutput> {
+        const searchDirectory = this.getSearchDirectory(this.fsPath)
+        try {
+            const results = await this.executeRipgrep(updates)
+            return this.createOutput(results)
+        } catch (error: any) {
+            this.logger.error(`Failed to search in "${searchDirectory}": ${error.message || error}`)
+            throw new Error(`Failed to search in "${searchDirectory}": ${error.message || error}`)
+        }
+    }
+
+    private getSearchDirectory(fsPath?: string): string {
+        const workspaceFolders = vscode.workspace.workspaceFolders
+        const searchLocation = fsPath
+            ? fsPath
+            : !workspaceFolders || workspaceFolders.length === 0
+              ? ''
+              : workspaceFolders[0].uri.fsPath
+        return searchLocation
+    }
+
+    private async executeRipgrep(updates?: Writable): Promise<string> {
+        const searchDirectory = this.getSearchDirectory(this.fsPath)
+        return new Promise(async (resolve, reject) => {
+            const args: string[] = []
+
+            // Add search options
+            if (!this.caseSensitive) {
+                args.push('-i') // Case insensitive search
+            }
+            args.push('--line-number') // Show line numbers
+
+            // No heading (don't group matches by file)
+            args.push('--no-heading')
+
+            // Don't use color in output
+            args.push('--color', 'never')
+
+            // Add include/exclude patterns
+            if (this.includePattern) {
+                // Support multiple include patterns
+                const patterns = this.includePattern.split(',')
+                for (const pattern of patterns) {
+                    args.push('--glob', pattern.trim())
+                }
+            }
+
+            if (this.excludePattern) {
+                // Support multiple exclude patterns
+                const patterns = this.excludePattern.split(',')
+                for (const pattern of patterns) {
+                    args.push('--glob', `!${pattern.trim()}`)
+                }
+            }
+
+            // Add search pattern and path
+            args.push(this.query, searchDirectory)
+
+            this.logger.debug(`Executing ripgrep with args: ${args.join(' ')}`)
+
+            const options: ChildProcessOptions = {
+                collect: true,
+                logging: 'yes',
+                rejectOnErrorCode: (code) => {
+                    if (code !== 0 && code !== 1) {
+                        this.logger.error(`Ripgrep process exited with code ${code}`)
+                        return new Error(`Ripgrep process exited with code ${code}`)
+                    }
+                    return new Error()
```

Comment:
Can we add some message for this error?

Reply:
shouldn't throw error when the code equal to 0 and 1(no matching), removed this error.